peer/peer.go

```diff
@@ -2076,6 +2091,36 @@ func (p *Peer) AssociateConnection(conn net.Conn) {
 	}(p)
 }
 
+// SetIsWhitelisted updates the whitelisted status of the peer based on
+// the provided peer whitelist.
+//
+// This should only be called after AssociateConnection ensuring the peer
+// references its underlying connection.
+func (p *Peer) SetIsWhitelisted(whitelist []*net.IPNet) {
```

Comment:
Per above, I don't understand why this would be here.  The ban manager is the thing that needs to control whitelisting.